src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs

```diff
@@ -1,23 +1,508 @@
+
+        /// <summary>
+        /// Gets a command that opens the pull request on GitHub.
+        /// </summary>
+        public ReactiveCommand<object> OpenOnGitHub { get; }
+
+        /// <summary>
+        /// Gets a command that toggles the <see cref="ChangedFilesView"/> property.
+        /// </summary>
+        public ReactiveCommand<object> ToggleChangedFilesView { get; }
+
+        /// <summary>
+        /// Gets a command that toggles the <see cref="OpenChangedFileAction"/> property.
+        /// </summary>
+        public ReactiveCommand<object> ToggleOpenChangedFileAction { get; }
+
+        /// <summary>
+        /// Gets a command that opens a <see cref="IPullRequestFileViewModel"/>.
+        /// </summary>
+        public ReactiveCommand<object> OpenFile { get; }
+ 
+        /// <summary>
+        /// Gets a command that diffs a <see cref="IPullRequestFileViewModel"/>.
+        /// </summary>
+        public ReactiveCommand<object> DiffFile { get; }
+
+        /// <summary>
+        /// Initializes the view model with new data.
+        /// </summary>
+        /// <param name="data"></param>
         public override void Initialize([AllowNull] ViewWithData data)
         {
-            System.Windows.MessageBox.Show(String.Format(CultureInfo.InvariantCulture, "{0}", data.Data));
+            var prNumber = (int)data.Data;
+
+            IsBusy = true;
+
+            // TODO: Catch errors.
+            Observable.CombineLatest(
+                    repositoryHost.ApiClient.GetPullRequest(repository.Owner, repository.CloneUrl.RepositoryName, prNumber),
+                    repositoryHost.ApiClient.GetPullRequestFiles(repository.Owner, repository.CloneUrl.RepositoryName, prNumber).ToList(),
+                    (pr, files) => new { PullRequest = pr, Files = files })
+                .ObserveOn(RxApp.MainThreadScheduler)
+                .Subscribe(x => Load(x.PullRequest, x.Files).Forget());
+        }
+
+        /// <summary>
+        /// Loads the view model from octokit models.
+        /// </summary>
+        /// <param name="pullRequest">The pull request model.</param>
+        /// <param name="files">The pull request's changed files.</param>
+        public async Task Load(PullRequest pullRequest, IList<PullRequestFile> files)
+        {
+            model = pullRequest;
+            State = CreatePullRequestState(pullRequest);
+            SourceBranchDisplayName = GetBranchDisplayName(pullRequest.Head.Label);
+            TargetBranchDisplayName = GetBranchDisplayName(pullRequest.Base.Label);
+            CommitCount = pullRequest.Commits;
+            Title = pullRequest.Title;
+            Number = pullRequest.Number;
+            Author = new Models.Account(pullRequest.User, avatarProvider.GetAvatar(new AccountCacheItem(pullRequest.User)));
+            CreatedAt = pullRequest.CreatedAt;
+            Body = !string.IsNullOrWhiteSpace(pullRequest.Body) ? pullRequest.Body : "*No description provided.*";
```

Comment:
Yes, I did consider this but then we'd lose out on change notifications, as models shouldn't be implementing `INotifyPropertyChanged`. I could however use the `IPullRequestModel` fields as the backing instead of having them as fields in the VM itself.